server/backends/disk_cache/disk_cache.go

```diff
@@ -142,6 +154,27 @@ type DiskCache struct {
 	remoteInstanceName string
 }
 
+func Register(env environment.Env) error {
+	if *rootDirectory == "" || env.GetCache() != nil {
+		return nil
+	}
+	c, err := NewDiskCache(
+		env,
+		&config.DiskConfig{
+			RootDirectory:     *rootDirectory,
+			Partitions:        partitions,
+			PartitionMappings: partitionMappings,
+			UseV2Layout:       *useV2Layout,
+		},
+		cache_config.MaxSizeBytes(),
+	)
```

Comment:
nit: i think this formatting is kinda weird -- maybe cleaner as:

dc := &config.DiskConfig{
			RootDirectory:     *rootDirectory,
			Partitions:        partitions,
			PartitionMappings: partitionMappings,
			UseV2Layout:       *useV2Layout,
		}
c, err := NewDiskCache(env, dc, cache_config.MaxSizeBytes())